### PR TITLE
Distribution: Fix unable to bootstrap when an environment requires non-gcc/clang compilers

### PR DESCRIPTION
**Pull request by jnhealy2**
We ran into a problem creating a bootstrap mirror from an environment containing the following:
```
packages:
    c:
        require: intel-oneapi-compilers
    cxx:
        require: intel-oneapi-compilers
```
because clingo requires a gcc/clang compiler to build.  This change adds a fallback so that if the bootstrap mirror creation fails, it is re-attemepted in a temporary environemnt that does not contain any user-specified information.  This fix relies on spack finding a gcc compiler on the system that can be used to build clingo.

### CODE REVIEW

File: tests/test_distribution.py [hunk 1]
```diff
@@ -909,6 +910,37 @@ def __init__(self, _, cmd, args):
             parser.parse_args(call)
 
 
+def test_DistributionPackager_configure_bootstrap_mirror_fallback_to_empty_env(
+    tmpdir, monkeypatch
+):
+    """
+    This test verifies that `configure_bootstrap_mirror` does not construct a call to
+    `spack bootstrap` that violates its API.
```
Comment (sebrowne) [inline tests/test_distribution.py]: Can you be a bit more specific?  I think this is valid, but it seems like the primary purpose of the test is more like `"""Test that a failure creating the bootstrap mirror falls back to using an empty environment (generally this has been caused by overly-restrictive compiler settings, which don't really apply for the bootstrap mirror)"""`

Comment (jnhealy2) [reply tests/test_distribution.py]: done

**Review by sebrowne — APPROVED** [review]
